Install all dependencies using Bower

afeld opened the pull request:
Admittedly this adds a lot of junk to the repo, so am also considering using `.gitignore` to exclude the files we're not using.  Let me know if you feel strongly one way or another.

afeld committed:
install all dependencies using bower

afeld commented:
Should also note that I'm not installing Modernizr w/ Bower, since we're using a custom build.

haacked commented:
I have no thoughts on this. What is a bower? :) /cc @benbalter

benbalter commented:
All for Bower (@haacked Bower is an npm-style package manager for static assets), but as an idiosyncrasy, and to enforce pretty URLs for end-users, I'd prefer to set the path to something like `/vendor` via `.bowerrc`. Take a look at Render for an example.

I'm fine with not .gitignoring the various build components, but feel empowered to go for it if you've got a free cycle.

benbalter commented:
> Should also note that I'm not installing Modernizr w/ Bower, since we're using a custom build.

Maintainability > being a special snowflake. (assume we just didn't use the components we didn't need?) Feel free to bower if that's the case.

haacked commented:
Ok, I'll let @benbalter give the squirrel in that case. I'm cool with it. :smile:

afeld committed:
Merge remote-tracking branch 'origin/gh-pages' into bower

Conflicts:
	_includes/css/jquery.qtip.min.css
	_includes/header.html
	bower_components/qtip2/jquery.qtip.min.css
	css/jquery.qtip.min.css
	javascripts/jquery.qtip.min.js

afeld commented on css/application.css:
```diff
@@ -2,9 +2,6 @@
 ---
 
-{% include css/normalize.min.css %}
-{% include css/jquery.qtip.min.css %}
```
Ran into problems trying to reconcile `_includes` w/ Bower putting everything under one directory, so needed to undo this.  Sorry @XhmikosR!

XhmikosR replied:
No problem, it's 3 http requests instead of 1 so it's your loss :P

afeld replied:
Consequence accepted :wink:

afeld committed:
ignore test directories from build

should fix HTML::Proofer build errors:
https://travis-ci.org/github/choosealicense.com/builds/15327469

XhmikosR commented:
Are you sure the gain justifies the changes? I mean, it's so much bloat for just a few files...

afeld commented:
@XhmikosR Well, this was intended to make our lives easier, via
- specification of dependencies all in one place
- easy upgrading of libs

but with the various hassles it is causing, I'm now going to agree that it's not worth it and close this out.  I'll keep the branch around for a bit in case we feel reinvigorated about it.  Thanks for the feedback!

XhmikosR commented:
No problem, just sharing my thoughts :)

BTW, we should make what I did for the css with the js files too.

DoNotSpamPls commented:
`keep the branch around for a bit in case we feel reinvigorated about it`

4 (almost 5) years later 🤔
